### PR TITLE
Updated developer documentation - upgrading Go

### PR DESCRIPTION
**Pull request by alejandrox1**
Signed-off-by: Jorge Alarcon Ochoa <alarcj137@gmail.com>

### CODE REVIEW

File: contributors/devel/development.md [hunk 1]
```diff
@@ -150,9 +151,8 @@ images.
 
 - The image for cross compiling in [build/build-image/cross].
   The `VERSION` file and `Dockerfile`.
-- Update the desired Go version in Dockerfile for the [e2e][e2e-image] and [test][test-image].
-  This requires pushing the [e2e][e2e-image] and [test][test-image] images that are `FROM` the desired Go version.
```
Comment (cblecker) [inline contributors/devel/development.md]: The test image in `test/images/Makefile` needs to be updated too
https://github.com/kubernetes/kubernetes/pull/74632/files

Comment (alejandrox1) [reply contributors/devel/development.md]: Thank you for the ctach @cblecker !
Fixed it

**Pull request merged**